src/number_theory/number_field/embeddings.lean

```diff
@@ -123,3 +124,170 @@ end
 end bounded
 
 end number_field.embeddings
+
+namespace number_field
+
+section place
+
+variables {A : Type*} [normed_division_ring A] {K : Type*} [field K] (φ : K →+* A)
+
+/-- An embedding into a normed division ring defines a place of `K` -/
+def place : K → ℝ := norm ∘ φ
+
+lemma places.nonneg (x : K) : 0 ≤ place φ x := by simp only [place, norm_nonneg]
+
+@[simp]
+lemma places.eq_zero_iff (x : K) : place φ x = 0 ↔ x = 0 :=
+by simp only [place, norm_eq_zero, map_eq_zero]
+
+@[simp]
+lemma places.map_zero : place φ 0 = 0 :=
+by simp only [place, function.comp_app, map_zero, norm_zero]
+
+@[simp]
+lemma places.map_one : place φ 1 = 1 :=
+by simp only [place, function.comp_app, map_one, norm_one]
+
+@[simp]
+lemma places.map_inv (x : K) : place φ (x⁻¹) = (place φ x)⁻¹ :=
+by simp only [place, function.comp_app, norm_inv, map_inv₀]
+
+@[simp]
+lemma places.map_mul (x y : K) : place φ (x * y) = (place φ x) * (place φ y) :=
```

Comment:
You should definitely bundle as much as these properties as possible. `place : K →+* ℝ` will give some of them, you can even do `place : K →+* ℝ≥0` if you want, but I am not sure this is convenient, since `ℝ≥0` is a subtype.

Reply:
I have tried using `place : K →+* ℝ≥0` but it was quite inconvenient indeed to have a `subtype`

Reply:
> You should definitely bundle as much as these properties as possible. 

You mean define `place` as a structure with all these properties included? I can try but then I'll need something like `has_coe_to_fun`. I have tried to do that with `infinite_place` but I am not sure I did it properly. For example, I cannot get `congr_fun` to work directly when I have an equality of infinite places (I need to coerce them into functions first).

Reply:
Well,I think you mean redefine it as an absolute value. I can try that indeed.

Reply:
Yes, sorry, addition is not respected of curse. In practice you have already done it.

Reply:
Yes, I was able to do that. My problem now is that there is : no extensionality rule found for absolute_value, so I do not know how to prove equalities like `place (conjugate φ) = place φ`

Reply:
```lean
example {R S : Type*} [semiring R] [ordered_semiring S] {A₁ A₂ : absolute_value R S}
  (H : ∀ x, A₁ x = A₂ x) : A₁ = A₂ :=
fun_like.ext A₁ A₂ H
```
works for me, but I agree that `absolute_value` is missing an `@[ext]` attribute (you can add it if you want).

Reply:
#17883

( I copied the one for ring hom ;)